tests/optimizers/botorch_base_test.py

```diff
@@ -166,6 +166,56 @@ def test_find_optimum_in_1d_discrete_space(seed):
     ), "After figuring out the best of the three points, it should only propose that."
 
 
+def test_propose_random_until_enough_evaluations_without_missing_objective_values(seed):
+    space = ps.ParameterSpace()
+    space.add(ps.IntegerParameter("integ", (0, 2)))
+    batch_shape = torch.Size()
+
+    opt = SingleObjectiveBOTorchOptimizer(
+        search_space=space,
+        objective=Objective("loss", greater_is_better=False),
+        model=SingleTaskGP(
+            torch.empty((*batch_shape, 0, len(space)), dtype=torch.float64),
+            torch.empty((*batch_shape, 0, 1), dtype=torch.float64),
+        ),
+        acquisition_function_factory=partial(
+            UpperConfidenceBound, beta=1.0, maximize=False
+        ),
+        num_initial_random_samples=2,
+        max_pending_evaluations=1,
+        seed=seed,
+    )
+
+    es = opt.generate_evaluation_specification()
+    assert not es.optimizer_info[
+        "model_based_pick"
+    ], "No evaluation reported, 0 < 2 initial random samples"
+    opt.report(
+        es.create_evaluation(objectives={"loss": es.configuration["integ"] ** 2}),
+    )
+
+    es = opt.generate_evaluation_specification()
+    assert not es.optimizer_info[
+        "model_based_pick"
+    ], "One evaluation reported, 1 < 2 initial random samples"
+    opt.report(
+        es.create_evaluation(objectives={"loss": None}),
+    )
+
+    es = opt.generate_evaluation_specification()
+    assert not es.optimizer_info[
+        "model_based_pick"
+    ], "One valid evaluation reported, 1 < 2 initial random samples"
```

Comment:
Before this PR, this would have already been a model based pick.

Reply:
True, but it was consistent with the behavior described in the in-line comment.
But I like this solution now much better.